sphinx/source/docs/first_steps/first_steps_7.rst

```diff
@@ -22,12 +22,14 @@ To customize the file Bokeh creates for your visualization, import and call the
 |output_file| function. ``output_file()`` accepts various file-related
 arguments. For example:
 
-* ``filename``: the filename for the HTML file. If you don't supply a file name,
-  Bokeh will use the file name of your python file as a filename for the HTML
-  file it creates.
+* ``filename``: the filename for the HTML file
 * ``title``: the title for you document (to be used in the HTML's ``<title>``
   tag)
 
+If you don't use the ``output_file()`` function to define a custom file name,
+Bokeh defaults to using the file name of your Python file as a filename for the
```

Comment:
This is only valid for standalone mode. In jupyter notebooks it generates a random filename.

Reply:
It has nothing to do with standalone output. If you are running from a `*python file` e.g. `foo.py` then Bokeh is able to glean the filename and use it to generate `foo.html`. But if you are running from a jupyter notebook (which is not a python file) then no filename is available, so it falls back to a random filename (all it can do). In both cases, standalone (pure HTML+JS) output is generated. This is another reason why explicit `save` is probably always preferable in notebooks.

Reply:
In any case I would suggest something like 
> Bokeh tries to use the name of the currently running Python script for the filename of the HTML output. If a script filename is not available (for instance, in a  Jupyter notebook), then Bokeh will generate a random filename.

cc @tcmetzger

Reply:
> It has nothing to do with standalone output. If you are running from a _python file_ e.g. `foo.py` then Bokeh is able to glean the filename and use it to generate `foo.html`. But if you are running from a jupyter notebook (which is not a python file) then no filename is available, so it falls back to a random filename (all it can do). In both cases, standalone output (pure HTML+JS) is what is generated. This is another reason why explicit `save` is probably always preferable in notebooks.

Maybe I used the wrong words. By 'standalone' I mean non-jupyter mode. Is there a name for it?

It is not that impossible to get the name of the current file in jupyter notebooks (eg [stackoverflow]( https://stackoverflow.com/questions/12544056/how-do-i-get-the-current-ipython-jupyter-notebook-name)), but there's really no need to: `output_file` and `save` should be enough.

The new wording looks perfect to me.

Reply:
Wording updated with PR https://github.com/bokeh/bokeh/pull/12260